### ISSUE TITLE
defercheck produces false positives with nested closures

### ISSUE THREAD

**Issue by tamird** [opennota/check#22]
running `defercheck` on [CockroachDB](https://github.com/cockroachdb/cockroach) results in:
```
$ defercheck ./...
./config: config/testutil.go:48:3: Repeating defer testingLock.Unlock() inside function TestingSetupZoneConfigHook
./kv: kv/txn_coord_sender.go:688:3: Repeating defer tc.cleanupTxn() inside function updateState
```

These are both false positives, but this issue is about the first one. Here's the relevant code: https://github.com/cockroachdb/cockroach/blob/e02b984b0d3b7d50e290151d87175a724c467852/config/testutil.go#L33:L58

```go
// TestingSetupZoneConfigHook initializes the zone config hook
// to 'testingZoneConfigHook' which uses 'testingZoneConfig'.
// Settings go back to their previous values when the stopper runs our closer.
func TestingSetupZoneConfigHook(stopper *stop.Stopper) {
	testingLock.Lock()
	defer testingLock.Unlock()
	if testingHasHook {
		panic("TestingSetupZoneConfigHook called without restoring state")
	}
	testingHasHook = true
	testingZoneConfig = map[uint32]*ZoneConfig{}
	testingPreviousHook = ZoneConfigHook
	ZoneConfigHook = testingZoneConfigHook
	testingLargestIDHook = func() (max uint32) {
		testingLock.Lock()
		defer testingLock.Unlock()
		for id := range testingZoneConfig {
			if id > max {
				max = id
			}
		}
		return
	}

	stopper.AddCloser(stop.CloserFn(testingResetZoneConfigHook))
}
```
